x/auth/keeper/account.go

```diff
@@ -31,28 +33,30 @@ func (ak AccountKeeper) GetAccount(ctx sdk.Context, addr sdk.AccAddress) exporte
 	if bz == nil {
 		return nil
 	}
-	acc := ak.decodeAccount(bz)
-	return acc
+
+	return ak.decodeAccount(bz)
 }
 
 // GetAllAccounts returns all accounts in the accountKeeper.
 func (ak AccountKeeper) GetAllAccounts(ctx sdk.Context) (accounts []exported.Account) {
-	ak.IterateAccounts(ctx,
-		func(acc exported.Account) (stop bool) {
-			accounts = append(accounts, acc)
-			return false
-		})
+	ak.IterateAccounts(ctx, func(acc exported.Account) (stop bool) {
+		accounts = append(accounts, acc)
+		return false
+	})
+
 	return accounts
 }
 
 // SetAccount implements sdk.AccountKeeper.
 func (ak AccountKeeper) SetAccount(ctx sdk.Context, acc exported.Account) {
 	addr := acc.GetAddress()
 	store := ctx.KVStore(ak.key)
-	bz, err := ak.cdc.MarshalBinaryBare(acc)
+
+	bz, err := ak.cdc.MarshalAccount(acc)
 	if err != nil {
 		panic(err)
```

Comment:
Would there be value in returning the error here or would that be a larger refactor?

Reply:
We could return an error here, yes. I don't think it'll be that large of a refactor. However, the upstream caller would end up panicing anyway.